lib/core/include/irods/connection_pool.hpp

```diff
@@ -15,6 +16,33 @@
 
 namespace irods
 {
+    /// Defines various options for changing the behavior of a connection pool.
+    ///
+    /// \since 4.3.1
+    struct connection_pool_options
+    {
+        /// Defines the number of API requests that can be sent via a connection before
+        /// refreshing it.
+        ///
+        /// Independent of any other option.
+        ///
+        /// \since 4.3.1
+        std::optional<std::int16_t> refresh_connections_after_n_retrievals;
+
+        /// Defines the number of seconds that must pass before refreshing a connection.
+        ///
+        /// Independent of any other option.
+        ///
+        /// \since 4.3.1
+        //std::optional<std::chrono::seconds> refresh_connections_after_n_seconds;
```

Comment:
Remove or replace the refresh_time parameter in the constructor of a connection pool.

Reply:
I'm considering moving the `refresh_time` constructor parameter with this. Doing that means all existing constructors become deprecated, but we get a consistent interface.

Reply:
that seems good - consolidation is fine.  the connection pool is still relatively new, and we have written most of the clients using them.

Reply:
Will add words explaining how the options play together.